### ISSUE TITLE
Bug Report: Issue with `next-css-obfuscator` in Next.js 15.0.0 Using `app router`

### ISSUE THREAD

**Issue by nhannt201** [soranoo/next-css-obfuscator#48]
# Bug Report: `next-css-obfuscator` Causes Unexpected `SyntaxError` in Next.js 15.0.0 with Middleware and Obfuscation

## Type
- [ ] Incorrect Class Name Extraction
- [x] Incorrect JavaScript Obfuscation
- [ ] Incorrect HTML Obfuscation
- [ ] Incorrect CSS Obfuscation
- [ ] Others

---

## Checklist
1. [x] Updated the package to the latest version.
2. [x] Read all documentation.
3. [x] No related issue found.
4. [x] Meaningful issue title provided.

---

## Environment
- **Package Version**: v0.0.0
- **Node Version**: v20.x.x
- **Next.js Version**: 15.0.0
- **Others**: Using `app router` with dynamic routing and middleware.

---

## Describe the Bug
When applying `next-css-obfuscator` to a Next.js 15.0.0 project, dynamic routes using middleware and app directory structure fail with a `SyntaxError: Unexpected strict mode reserved word`. 

The error occurs specifically in:
```javascript
originalRequire.call(this, request);
```

### Observed Behavior
- On the client-side, the application displays a `500 Internal Server Error`.
- Server-side logs show:
  ```
  ⨯ SyntaxError: Unexpected strict mode reserved word
      at <unknown> (...\.next\server\app\(customPage)\page.js.js:1) 
      at mod.require (\node_modules\next\dist\server\require-hook.js:65:28)
      at requirePage (\node_modules\next\dist\server\require.js:103:84)
      at loadComponentsImpl (\node_modules\next\dist\server\load-components.js:100:57)
  ```
- The issue seems related to obfuscation changing the syntax in strict mode.

---

## To Reproduce
1. Create a Next.js app using version 15.0.0.
2. Configure middleware to redirect `/page/[pageId]` to `src/app/(customPage)/page.tsx`.
3. Apply `next-css-obfuscator` to the project with the following configuration:
    ```javascript
    // next-css-obfuscator.config.cjs
    module.exports = {
      enable: true,
      mode: "random",
      refreshClassConversionJson: false,
      allowExtensions: [".jsx", ".tsx", ".js", ".ts", ".html", ".rsc"],
      blackListedFolderPaths: [
        "./.next/cache",
        /\.next\/server\/pages\/api/,
        /_document..*js/,
        /_app-.*/,
        /__.*/, // <= helpful for Local Fonts
      ],
    };
    ```
4. Run the app and navigate to a dynamic route, e.g., `/page/123`.

---

## Expected Behavior
- Dynamic routes should render correctly, and middleware should work as expected.
- Obfuscation should not break syntax or interfere with Next.js runtime.

---

## Screenshots
N/A

---

## Configuration
```javascript
// next-css-obfuscator.config.cjs
module.exports = {
  enable: true,
  mode: "random",
  refreshClassConversionJson: false,
  allowExtensions: [".jsx", ".tsx", ".js", ".ts", ".html", ".rsc"],
  blackListedFolderPaths: [
    "./.next/cache",
    /\.next\/server\/pages\/api/,
    /_document..*js/,
    /_app-.*/,
    /__.*/, // <= helpful for Local Fonts
  ],
};
```

---

## Related Files
- **Dynamic Route File**: Located at `src/app/(customPage)/page.tsx`.
- **Middleware**: Handles redirection for `/page/[pageId]`.

### Dynamic Route Example
```javascript
import { notFound } from "next/navigation";
import { Metadata } from "next";

export async function generateMetadata({ searchParams }: any): Promise<Metadata> {
  return {
    title: "Dynamic Page",
    description: "This is a dynamically generated page.",
  };
}

const MyPage = async ({ searchParams }: { searchParams: any }) => {
  const { pageId } = searchParams;

  if (!pageId) {
    notFound();
  }

  return <PageUI pageId={pageId} />;
};

export default MyPage;
```

---

## Additional Context
- The issue might be related to how obfuscator modifies or interacts with dynamic imports, `require` hooks, and middleware in the new `app router`.
- Disabling `ppr = true` does not resolve the issue.
- Ignoring `src/app` in obfuscator settings also causes unrelated folders like `src/ui` to be skipped for obfuscation.

**Comment by soranoo** [comment]
An investigation will begin shortly (but most likely won't be in this month)

**Comment by soranoo** [comment]
@nhannt201 Thank you for sponsoring!

**Comment by soranoo** [comment]
I don't see any problem with my [reproduction](https://github.com/soranoo/issues-playground/tree/main/next-css-obfuscator-48)

To run the reproduction,
`npm i && npm run build && npm run obs && npm run start`

**Comment by nhannt201** [comment]
I used an additional layout layer to wrap the children, and outside of the children and WrapperComponent, I used MUI combined with TailwindCSS for handling. However, I encountered the above error. The issue is that sometimes it occurs in my project, and sometimes it doesn't. I also tried to reproduce it again, but I haven't been successful so far. Perhaps I will close the issue here and reopen it if I can reproduce the error along with the accompanying source. Thank you for your support.

But it seems like the className from MUI UI or other libraries cannot be transformed, except for modules with TailwindCSS.
